algorithms/src/snark/marlin/ahp/ahp.rs

```diff
@@ -279,17 +277,18 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         end_timer!(v_X_at_beta_time);
 
         let z_b_s_at_beta = z_b_s
-            .iter()
-            .map(|(circuit_id, z_b_i)| {
-                let z_b_i_s = z_b_i.iter().map(|z_b| evals.get_lc_eval(z_b, beta).unwrap()).collect::<Vec<F>>();
-                (*circuit_id, z_b_i_s)
+            .values()
+            .map(|z_b_i| {
+                let z_b_i_s = z_b_i.iter().map(|z_b| evals.get_lc_eval(z_b, beta)).collect::<Result<Vec<F>, _>>();
+                z_b_i_s
             })
-            .collect::<BTreeMap<_, _>>();
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Ditto here, I think we can use `try_collect` to still collect into a BTreeMap

Reply:
I can't use `try_collect` because I need to annotate the types, but I can use `?` within the map now and collect: 4916a3ca0